web/gds-user-ui/src/components/Contacts/ContactsForm.tsx

```diff
@@ -1 +1,94 @@
-export {};
+import { FormProvider, useForm } from 'react-hook-form';
+import { t } from '@lingui/macro';
+import { chakra } from '@chakra-ui/react';
+import StepButtons from 'components/StepsButtons';
+import ContactForm from 'components/Contacts/ContactForm';
+import useCertificateStepper from 'hooks/useCertificateStepper';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { contactsValidationSchema } from 'modules/dashboard/certificate/lib/contactsValidationSchema';
+import { StepEnum } from 'types/enums';
+import { useFetchCertificateStep } from 'hooks/useFetchCertificateStep';
+import { useUpdateCertificateStep } from 'hooks/useUpdateCertificateStep';
+import FormLayout from 'layouts/FormLayout';
+import MinusLoader from 'components/Loader/MinusLoader';
+
+const ContactsForm: React.FC = () => {
+    const { previousStep, nextStep, currentState } = useCertificateStepper();
+    const { certificateStep, isFetchingCertificateStep } = useFetchCertificateStep({
+    key: StepEnum.CONTACTS
+    });
+
+    const { updateCertificateStep, updatedCertificateStep } = useUpdateCertificateStep();
+
+    const resolver = yupResolver(contactsValidationSchema);
+
+    const methods = useForm({
+    defaultValues: certificateStep?.form,
+    resolver,
+    mode: 'onChange'
+  });
+
+    const {
+    formState: { isDirty }
+    } = methods;
+
+    const handleNextStepClick = () => {
+        console.log('[] handleNextStep', methods.getValues());
+        // if the form is dirty, then we need to save the data and move to the next step
+        console.log('[] isDirty', isDirty);
+        if (!isDirty) {
+          console.log('[] is not Dirty', isDirty);
+          nextStep(updatedCertificateStep?.errors ?? certificateStep?.errors);
+        } else {
+          const payload = {
+            step: StepEnum.CONTACTS,
+            form: {
+              ...methods.getValues(),
+              state: currentState()
+            } as any
+          };
+          console.log('[] isDirty  payload', payload);
+    
+          updateCertificateStep(payload);
+          console.log('[] isDirty 3 (not)', updatedCertificateStep);
+          nextStep(updatedCertificateStep?.errors);
+        }
+      };
+    
```

Comment:
Would you mind adding the handlePreviousStepClick function too ? I just add it to the legal form too n my current PR.
https://github.com/trisacrypto/directory/blob/663d0cdfc2b6b352849012d0459181198a730975/web/gds-user-ui/src/components/LegalPerson/LegalForm.tsx